k8s/pkg/apis/longhorn/v1beta1/engine.go

```diff
@@ -126,6 +126,7 @@ type EngineSpec struct {
 type EngineStatus struct {
 	InstanceStatus `json:""`
 
+	// +kubebuilder:validation:Type=string
```

Comment:
Why is the current size type string rather than integer?

Reply:
The json tag specify to use string rather than integer. So, that's why we need to enforce the type with string.

I did a test, without specific to use string, it failed on json unmarshal when call List API.

Reply:
Not sure if removing suffix `,string` from below json tag `json:"currentSize,string"` work or not.

Reply:
Seems no https://play.golang.org/p/hJifNZNR0l6

Reply:
This link is unavailable...

Reply:
Well, it's available to me. Never mind, I did a quick test, it's not working.
```
package main

import (
	"encoding/json"
	"fmt"
)

type EngineStatus struct {
	CurrentSize int64 `json:"currentSize"`
}

func main() {
	es := EngineStatus{}
	data := []byte(`{"currentSize": "100"}`)

	json.Unmarshal(data, &es)
	fmt.Printf("%v\n", es.CurrentSize)
}
```

Reply:
Shouldn't the test be like this?
```
package main

import (
	"encoding/json"
	"fmt"
)

type EngineStatus struct {
	CurrentSize int64 `json:"currentSize"`
}

func main() {
	es := EngineStatus{}
	data := []byte(`{"currentSize": 100}`)

	json.Unmarshal(data, &es)
	fmt.Printf("%v\n", es.CurrentSize)
}
```

This test works fine. `// +kubebuilder:validation:Type=int64` indicates the input data value is an integer. `json:"currentSize"` means the data value type is not required to be string.


But once you modify the definition to `CurrentSize int64 `json:"currentSize,string"``, this does not work anymore.

Reply:
But the existing `status.currentSize` in CR is a string type.

Reply:
You mean the existing CR field is stored as a string type in ETCD? If that's the case, it's fine to leave it here. Or we can modify this part during the CRD API version bump.

Reply:
Yes, existing CR field is stored as a string type in ETCD.

Agree, we could handle it in the v1beta2. I'll create an issue for this.

Reply:
Sure. And you can add the issue into this Epic:
https://github.com/longhorn/longhorn/issues/3276

Reply:
Or we could wait the PR https://github.com/longhorn/longhorn-manager/pull/1136 be merged, and we focus on CRD structural schema for v1beta2 version only.

Then, we could change the struct from string to int, and implement the conversion in upgrade path.

Reply:
There is no need to wait for it... We still need to open another PR to remove all deprecated fields after https://github.com/longhorn/longhorn-manager/pull/1136 merge.

Reply:
Okay, new [issue](https://github.com/longhorn/longhorn/issues/3282) created.